### DISCUSSION TITLE
Sending Button With Url

### DISCUSSION THREAD

**Discussion by xirf** [open-wa/wa-automate-nodejs#2392] [Q&A] [answered]
How can i send button with url?

**Comment by smashah** [comment]
URL & Call buttons were previously not possible because the host account device would scrub/mutate the message payload. This should now be possible for MD accounts and will be implemented some time this month.

Please see #98 for some background context.

Track #2394 for progress.

When this is implemented I will edit this post with instructions in the main comment.

Thanks

**Comment by smashah** [comment, marked as answer]
@evaasmakula 

You can use `B` (insiders) to send custom payloads to the internal socket.

```
await client.B(chatId, {
    "templateMessage": {
        "hydratedTemplate": {
            "hydratedButtons": [
                {
                    "index": 1,
                    "urlButton": {
                        "displayText": "I am a link",
                        "url": "https://github.com/open-wa/wa-automate-nodejs"
                    }
                },
                {
                    "index": 2,
                    "callButton": {
                        "displayText": "Call me now!",
                        "phoneNumber": "+1234567890"
                    }
                },
                {
                    "index": 3,
                    "quickReplyButton": {
                        "displayText": "Or just click me",
                        "id": "button-1"
                    }
                }
            ],
            "hydratedContentText": "content",
            "hydratedFooterText": "footerer"
        }
    }
});
```

result:

![image](https://user-images.githubusercontent.com/5903821/162698045-536aa7ed-1e26-4113-991d-5b19f70d8b07.png)


track: https://github.com/open-wa/wa-automate-nodejs/issues/2394
